internal/service/container/inspect.go

```diff
@@ -83,6 +85,47 @@ func (s *service) Inspect(ctx context.Context, cid string, sizeFlag bool) (*type
 	return &cont, nil
 }
 
+func getHostConfigFromDockerCompat(c *dockercompat.HostConfig) *types.ContainerHostConfig {
+	if c == nil {
+		return nil
+	}
+
+	hostConfigDevices := []types.DeviceMapping{}
+	for _, device := range c.Devices {
+		hostConfigDevices = append(hostConfigDevices, types.DeviceMapping{
+			PathOnHost:        device.PathOnHost,
+			PathInContainer:   device.PathInContainer,
+			CgroupPermissions: device.CgroupPermissions,
+		})
+	}
+
+	return &types.ContainerHostConfig{
+		ContainerIDFile: c.ContainerIDFile,
+		LogConfig: types.LogConfig{
+			Type:   c.LogConfig.Driver,
+			Config: c.LogConfig.Opts,
+		},
+		PortBindings:   c.PortBindings,
+		IpcMode:        c.IpcMode,
+		PidMode:        c.PidMode,
+		ReadonlyRootfs: c.ReadonlyRootfs,
+		ShmSize:        c.ShmSize,
+		Sysctls:        c.Sysctls,
+		CPUSetMems:     c.CPUSetMems,
+		CPUSetCPUs:     c.CPUSetCPUs,
+		CPUShares:      int64(c.CPUShares),
+		CPUPeriod:      int64(c.CPUPeriod),
+		Memory:         c.Memory,
+		MemorySwap:     c.MemorySwap,
+		OomKillDisable: c.OomKillDisable,
+		Devices:        hostConfigDevices,
+		// TODO: Uncomment these when https://github.com/runfinch/finch-daemon/pull/267 gets merged
+		// CPURealtimePeriod: inspect.HostConfig.CPURealtimePeriod,
+		// CPURealtimeRuntime: inspect.HostConfig.CPURealtimeRuntime,
+		// TODO: add blkio devices which requires a change in the dockercompat response from Nerdctl
```

Comment:
Are CPURealtime* and Blkio* the only options left from the HostConfig?

Reply:
There are other fields that haven't been implemented in Nerdctl yet.